packages/EasyTest/src/Traits/MessengerAssertionsTrait.php

```diff
@@ -113,22 +112,31 @@ static function (object $message) use ($messageClass, $expectedProperties, $prop
             }
         );
 
-        Assert::assertCount($messagesCount ?? 1, $envelopes);
+        Assert::assertCount($messagesCount, $envelopes);
     }
 
-    public static function consumeAsyncMessages(): void
+    /**
+     * @param array<int, array<class-string<\Throwable>, int|string>> $expectedExceptions
+     */
+    public static function consumeAsyncMessages(array $expectedExceptions = []): void
     {
-        self::runMessengerWorker('async', 'default');
+        if (isset(self::$isInsideSafeCall) && self::$isInsideSafeCall) {
+            throw new RuntimeException(
+                "You can't use MessengerAssertionsTrait::consumeAsyncMessages() in ExceptionTrait::safeCall()"
```

Comment:
Could you please comment why we can't?

Reply:
Starting a worker should always work without exceptions.

If there are exceptions, we should not catch it.

Reply:
Let's suppose I expect an exception from the message handler and want to use `safeCall` for this. How should we handle this situation?

Reply:
Explanation added in doc comments.

Expected exceptions should be provided in the `$expectedExceptions` argument.